lib/pbench/test/unit/server/test_datasets_list.py

```diff
@@ -698,3 +699,76 @@ def test_key_summary(self, query_as):
                 "origin": None,
             },
         }
+
+    @pytest.mark.parametrize(
+        "sort,results",
+        [
+            (
+                "dataset.name",
+                ["fio_1", "fio_2", "test", "uperf_1", "uperf_2", "uperf_3", "uperf_4"],
+            ),
+            (
+                "dataset.name:asc",
+                ["fio_1", "fio_2", "test", "uperf_1", "uperf_2", "uperf_3", "uperf_4"],
+            ),
+            (
+                "dataset.name:desc",
+                ["uperf_4", "uperf_3", "uperf_2", "uperf_1", "test", "fio_2", "fio_1"],
+            ),
+            (
+                "dataset.uploaded",
+                ["test", "fio_1", "uperf_1", "uperf_2", "uperf_3", "uperf_4", "fio_2"],
+            ),
+            (
+                "dataset.uploaded:desc",
+                ["fio_2", "uperf_4", "uperf_3", "uperf_2", "uperf_1", "fio_1", "test"],
+            ),
+            (
+                "dataset.metalog.run.controller",
+                ["test", "fio_1", "fio_2", "uperf_1", "uperf_2", "uperf_3", "uperf_4"],
+            ),
+            (
+                "global.test.sequence:desc",
+                ["fio_1", "fio_2", "test", "uperf_1", "uperf_2", "uperf_3", "uperf_4"],
+            ),
+            (
+                "global.test.sequence",
+                ["uperf_4", "uperf_3", "uperf_2", "uperf_1", "test", "fio_2", "fio_1"],
+            ),
+            (
+                "user.test.odd,global.test.sequence:desc",
+                ["fio_1", "test", "uperf_2", "uperf_4", "fio_2", "uperf_1", "uperf_3"],
+            ),
+            (
+                "user.test.odd:desc,dataset.name:desc",
+                ["uperf_3", "uperf_1", "fio_2", "uperf_4", "uperf_2", "test", "fio_1"],
+            ),
+        ],
+    )
+    def test_dataset_sort(self, server_config, query_as, sort, results):
+        """Test `datasets/list?sort`
+
+        We want a couple of consistent values sequences to play with. We can
+        use the dataset.name and dataset.resource_id fields, but we want to
+        cross Metadata namespaces, so add "global" and "user" keys we can
+        order.
+
+        Args:
+            server_config: The PbenchServerConfig object
+            query_as: A fixture to provide a helper that executes the API call
+            login: The username as which to perform a query
+            query: A JSON representation of the query parameters (these will be
+                automatically supplemented with a metadata request term)
+            results: A list of the dataset names we expect to be returned
+        """
+
+        # Assign "sequence numbers" in the inverse order of name
+        test = User.query(username="test")
+        all = Database.db_session.query(Dataset).order_by(desc(Dataset.name)).all()
+        for i, d in enumerate(all):
+            odd = i & 1
+            Metadata.setvalue(d, "global.test.sequence", i)
```

Comment:
It would be a better test if `sequence` were totally unrelated to the alphabetical sequence of the names (because we already have a test scenario for that).  I would suggest using the `ord()` value of the second (or last) letter in the name, but the names are all too similar to each other...perhaps use the `hash()` of the name?

Reply:
I don't see how that would make any difference. If you're suggesting that SQL is somehow sorting on the wrong key and we can't detect it, I'm not sure there's much we could do to guarantee that. It seems overly paranoid. Plus, with this, I can look at the list and easily predict (and check) the order rather than generating and then manually comparing a list of hash strings.

If we're going to be that paranoid, I think we're kinda lost anyway.

Reply:
I totally understand (and sympathize with) the desire to be able to easily predict the search order by inspection.  But, having two searches produce the same ordering seems redundant to me (from a testing perspective).  If we peek inside the black box, we can convince ourselves that the two searches really do do different things, but the test doesn't really prove it...and, if the test scenario doesn't prove anything, shouldn't we omit it?

(The ordering wouldn't have to use `hash()` -- I just seized on that as an easy-to-implement alternative -- the problem is that the results are already heavily ordered, and so it's hard to find a new characteristic in them which doesn't produce one of the existing orders.  🥴  I briefly considered using a pseudo-randomly generated sequence with a known seed (which makes it reproducible), but I figured that just using `hash()` was easier....  😉)

(Just because you're not paranoid doesn't mean They aren't out to get you!  🙃)